addon/bootstrap.js

```diff
@@ -123,14 +114,20 @@ function shutdown(addonData, reason) {
     if (!studyUtils._isEnding) {
       // we are the first 'uninstall' requestor => must be user action.
       log.debug("probably: user requested shutdown");
-      studyUtils.endStudy({reason: "user-disable"});
-      return;
+      studyUtils.endStudy({ reason: "user-disable" });
+      // TODO: Why is the following return needed!?
+      // return;
```

Comment:
Yeah, it would be good to find out if you need the return or not. Maybe `endStudy` calls into this by ways of uninstalling the add-on?

Reply:
Good point. I dug into that and turns out it's to have different behaviours between uninstall and "addon disabled". Returning in that block prevents from uninstalling the add-on. We don't care about that, we're treating "disable" like uninstall in our study. I've dropped a comment there to clarify.